app/models/toggle_Enable_event.rb

```diff
@@ -0,0 +1,20 @@
+class ToggleEnableEvent < AggregatedEvent
+  self.abstract_class = true
+
+  belongs_to_aggregate :certificate
+  validates_presence_of :certificate
+  validate :certificate_is_signing?
+  after_save TriggerMetadataEventCallback.publish
+
+  def attributes_to_apply
+    { enabled: enabled }
+  end
+
+  private
+
+  def certificate_is_signing?
+    return unless certificate.usage == 'encryption'
```

Comment:
I think should probably be a check like:

```suggestion
    return if certificate.usage == 'signing'
```

Reply:
Alternatively, we could add some helpers to the `Certifcate` class like '#signing?' so that we can abstract away the data model.